### ISSUE TITLE
Proxy Environment variables are ignored

### ISSUE THREAD

**Issue by marczis** [awslabs/aws-sam-cli#330]
At lest seems to be the case for me with:
lambci/lambda       nodejs6.10

,If I create an env.json file with `http_proxy` and `https_proxy` with in, the nodejs code still can't reach AWS Services. 
On the other hand if I add this small snippet into my main js file:

```
//For SAM local
if (process.env["AWS_SAM_LOCAL"]) {
	console.log("=============> SaM LoCaL RuN <==============")
	const AWS = require('aws-sdk');
	const proxy = require('proxy-agent');
	AWS.config.update({
		httpOptions: { agent: proxy("http://MySecretproxy.com:8080") }
	});	
}
```

things starts to work as expected.

**Comment by marczis** [comment]
Looks like I can't add labels by myself, but I would consider this as a feature request instead of a issue, plus there is a working workaround.

**Comment by jfuss** [comment]
@marczis I can quite wrap my head around what is being asked here. Are you asking to be able to set env-vars on the local lambdas when running `sam local start-api`? So something like `sam local start-api -e env.json` which would replace the env vars just like `sam local invoke`?

If so this is addressed in our [0.3.0 release](https://github.com/awslabs/aws-sam-cli/releases/tag/v0.3.0)

**Comment by jfuss** [comment]
When you pass in environment variables into the command, the keys need to be in the Environment.Variables on the Function, otherwise they will not be added. 

This seems like a configuration issue with your template and we don't add arbitrary environment variables into the container. I am going to close this issue, as I don't believe this is a bug but feel free to reopen if I misunderstood something.
